core/src/main/java/aima/core/search/api/SearchForActionsBidirectionallyFunction.java

```diff
@@ -1,13 +1,15 @@
 package aima.core.search.api;
 
+import aima.core.util.datastructure.Pair;
+
 import java.util.List;
 import java.util.function.Function;
 
 /**
  * @author manthan
  */
 @FunctionalInterface
-public interface SearchForActionsBidirectionallyFunction<A, S> extends Function<BidirectionalProblem<A, S>, List<A>> {
+public interface SearchForActionsBidirectionallyFunction<A, S> extends Function<Pair<Problem<A, S>, Problem<A, S>>, List<A>> {
     @Override
-    List<A> apply(BidirectionalProblem<A, S> problem);
+    List<A> apply(Pair<Problem<A, S>, Problem<A, S>> problem);
```

Comment:
This should not take a pair, but broken apart as:

originalProblem, reverseProblem

Using Pair on the ProblemFactory was for convenience as you can't specific generics in an Array i.e. Problem[].